source/common/http/utility.cc

```diff
@@ -517,7 +517,10 @@ void Utility::sendLocalReply(const bool& is_reset, const EncodeFunctions& encode
 
   // Respond with a gRPC trailers-only response if the request is gRPC
   if (local_reply_data.is_grpc_) {
-    response_headers->setStatus(std::to_string(enumToInt(Code::OK)));
+    response_headers->setStatus(httpCodeForGrpcDeny(response_headers)
+                                    ? response_headers->getStatusValue()
```

Comment:
I feel like this should be using httpToGrpcStatus on the observed grpc-status from the authorization server. Either that, or I misunderstand the original intention.

Reply:
I missed that function when I made the first pass at this, but as @rojkov noted above, it's simpler to just only reset as ...OK if this option isn't set. Next push will have that.